pkg/controller/login/reset_password.go

```diff
@@ -79,6 +81,20 @@ func (c *Controller) HandleSubmitResetPassword() http.Handler {
 		var resetComposer auth.ResetPasswordEmailFunc
 
 		membership := controller.MembershipFromContext(ctx)
+
+		// This is likely - most users reset password from un-authed context.
+		if membership == nil {
```

Comment:
The middleware sets CurrentMemberships (plural) too - you can use that instead of doing a lookup.

Reply:
only for RequireAuth pages, which this is not. I don't believe the user will have a session (with currentmembership and realm set)

Reply:
It gets populated by LoadCurrentMembership, not requireauth. And I'm pretty sure this route calls LoadCurrentMembership

Reply:
These do:
https://github.com/google/exposure-notifications-verification-server/blob/008e6956679230a7c51085ef8724fbe9f00dae30/internal/routes/server.go#L182

this one is here: https://github.com/google/exposure-notifications-verification-server/blob/008e6956679230a7c51085ef8724fbe9f00dae30/internal/routes/server.go#L168

Reply:
Ah blarg. I messed up that ordering then. Do you have bandwidth to refactor the middleware to load memberships and require that middleware here?

Loading memberships is really expensive (it's an O(n+1) query), so I'd prefer to do it in middleware and cache it on the context once.

Reply:
This is unauthenticated (the user types in their email address) so there is not user or membership yet. I've change this to be less expensive by db querying without loading all memberships.